src/models/DefaultPluginData.js

```diff
@@ -0,0 +1,91 @@
+import { version as CORE_VERSION } from '../../package.json';
+import Component from './Component';
+
+/**
+ * Class that represents all data of a Plugin.
+ */
+class DefaultPluginData {
+  /**
+   * Default constructor.
+   * @param {String} props.name - Name of plugin.
+   * @param {String} props.version - Version of plugin.
+   */
+  constructor(props = {
+    name: null,
+    version: null,
+  }) {
+    /**
+     * Plugin name.
+     * @type {String}
+     */
+    this.name = props.name || null;
+    /**
+     * Plugin version.
+     * @type {String}
+     */
+    this.version = props.version || null;
+    /**
+     * All plugin components.
+     * @type {Component[]}
+     */
+    this.components = [];
+    /**
+     * All plugin links.
+     * @type {ComponentLink[]}
+     */
+    this.links = [];
+    /**
+     * All plugin definitions.
+     * @type {{components: ComponentDefinition[], links: ComponentLinkDefinition[]}}
+     */
+    this.definitions = {
+      components: [],
+      links: [],
+    };
+    /**
+     * All parser errors.
+     * @type {ParseError[]}
+     */
+    this.parseErrors = [];
+  }
+
+  /**
+   * Get version of plugin core.
+   * @returns {String} Version of plugin core.
+   */
+  get coreVersion() {
+    return CORE_VERSION;
+  }
+
+  /**
+   * Get component by id.
+   * @param {String} id - Component id.
+   * @return {Component} Component or undefined.
+   */
+  getComponentById(id) {
+    return this.components.find((component) => id === component.id);
```

Comment:
Clarification regarding the structure may be necessary here, drawer (as well as test cases) seems to assume a tree of ``[{..., children:[{...}]}]`` but both this method and removeComponentById seem to assume a flat array, being only able to search on one level.

Reply:
your right I forget this, thank's

Reply:
done